commands/snapshotautodeletecommand.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * @file
+ * @version 0.1
+ * @copyright 2017 CN-Consult GmbH
+ * @author Jens Stahl <jens.stahl@cn-consult.eu>
+ *
+ * License: Please check the LICENSE file for more information.
+ */
+
+namespace PBST\Command;
+
+use ProfitBricksApi\Snapshot;
+use ProfitBricksApi\VirtualMachine;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\TableSeparator;
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Helper\TableStyle;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use ProfitBricksApi\ProfitBricksApi;
+use DateTime;
+use Exception;
+
+/**
+ * Class SnapshotAutoDeleteCommand
+ *
+ * This class implements a simple automated deletion which is controlled by a time range defined in config.ini.
+ */
+class SnapshotAutoDeleteCommand extends Command
+{
+    private $config;
+    /** @var  ProfitBricksApi $profitBricksApi */
+    private $profitBricksApi;
+    /** @var null|VirtualMachine[] $virtualMachines */
+    private $virtualMachines = null;
+
+    protected function configure()
+    {
+        $this->profitBricksApi = new ProfitBricksApi();
+        $this
+            ->setName("snapshot:autoDelete")
+            ->setDescription("Deletes snapshots regarding to the configuration in config.ini!");
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if (is_readable('config.ini'))
+        {
+            $this->config = parse_ini_file('config.ini', true);
+            if (!isset($this->config['api']['user']) || !isset($this->config['api']['password'])) throw new Exception("No user or no password configured to connect ProfitBricks!");
+        }
+        else throw new Exception("Error during reading config.ini!");
+        $this->profitBricksApi->setUserName($this->config["api"]["user"]);
+        $this->profitBricksApi->setPassword($this->config["api"]["password"]);
```

Comment:
**It is no review request** but as I look for code duplication (as usual) and read this code block in each command, I am thinking about refactoring the marked block into an init-method of a (to be created) baseclass for all the commands.